forkify/src/js/model.js

```diff
@@ -0,0 +1,139 @@
+import { async } from "regenerator-runtime";
+import { API_URL, RES_PER_PAGE, KEY } from "./config.js";
+import { getJSON, sendJSON } from "./helpers.js";
+
+export const state = {
+  recipe: {},
+  search: {
+    query: "",
+    results: [],
+    page: 1,
+    resultsPerPage: RES_PER_PAGE,
+  },
+  bookmarks: [],
+};
+
+const createRecipe = (data) => {
+  let { recipe } = data.data;
+
+  return {
+    id: recipe.id,
+    title: recipe.title,
+    publisher: recipe.publisher,
+    sourceUrl: recipe.source_url,
+    image: recipe.image_url,
+    servings: recipe.servings,
+    cookingTime: recipe.cooking_time,
+    ingredients: recipe.ingredients,
+    key: recipe.key ? recipe.key : "",
+  };
+};
+
+export const loadRecipe = async function (id) {
+  try {
+    const data = await getJSON(`${API_URL}${id}?key=${KEY}`);
+    state.recipe = createRecipe(data);
+
+    state.recipe.bookmarked = state.bookmarks.some(
+      (bookmark) => bookmark.id === id
+    )
+      ? true
+      : false;
```

Comment:
Some returns boolean
![when-you-javascript-javascrıpt](https://user-images.githubusercontent.com/4405297/120087106-f4191a00-c0ed-11eb-9e69-04fbf6b2fe08.gif)